superset/assets/src/explore/components/controls/DateFilterControl.jsx

```diff
@@ -156,18 +163,24 @@ export default class DateFilterControl extends React.Component {
     this.setTypeCustomRange = this.setTypeCustomRange.bind(this);
     this.setTypeCustomStartEnd = this.setTypeCustomStartEnd.bind(this);
     this.toggleCalendar = this.toggleCalendar.bind(this);
+    this.changeTab = this.changeTab.bind(this);
   }
-  componentDidMount() {
+
+  componentWillMount() {
     const value = this.props.value;
     if (value.indexOf(SEPARATOR) >= 0) {
-      this.state = { ...this.state, ...getStateFromSeparator(value) };
+      this.setState({ ...this.state, ...getStateFromSeparator(value) });
```

Comment:
```
this.setState(getStateFromSeparator(value);
```

This will combine with existing state already. No need to combine with `this.state` manually.

Reply:
Oh thanks forgot to remove this. 🤦‍♀️